src/Mockery/Generator/StringManipulation/Pass/MethodDefinitionPass.php

```diff
@@ -70,8 +70,20 @@ protected function renderParams(Method $method, $config)
 
                     if (is_object($defaultValue)) {
                         $prefix = get_class($defaultValue);
-                        if ($isPhp81 && enum_exists($prefix)) {
-                            $prefix = var_export($defaultValue, true);
+                        if ($isPhp81) {
+                            if (enum_exists($prefix)) {
+                                $prefix = var_export($defaultValue, true);
+                            } elseif (
+                                !$param->isDefaultValueConstant() &&
+                                // "Parameter #1 [ <optional> F\Q\CN $a = new \F\Q\CN(param1, param2: 2) ]
+                                preg_match(
+                                    '#<optional>\s.*?\s=\snew\s(.*?)\s]$#',
+                                    $param->__toString(),
```

Comment:
It is not intended to ever call `__toString()` directly. This should be a cast instead (`(string)$param`).

Reply:
@Bilge I agree,

However the reason for using `__toString()` instead was because `$param` is actually an object that is proxying these calls via `__call` method to the `ReflectionParameter` which was giving an error because the `__toString` expects a string return type.

https://github.com/mockery/mockery/blob/b1be135c1ba7632f0248e07ee5e6e412576a309d/src/Mockery/Generator/Parameter.php#L28